src/tests/test_jsproxy.py

```diff
@@ -2100,3 +2100,59 @@ async def test_agen_aclose(selenium):
     assert await g.aclose() is None  # type:ignore[func-returns-value]
     p.destroy()
     assert l == ["finally"]
+
+
+@run_in_pyodide
+def test_gen_lifetimes(selenium):
+    import pytest
+
+    from pyodide.code import run_js
+    from pyodide.ffi import JsGenerator
+
+    f = run_js(
+        """
+        (function *(x) {
+            let l = [x];
+            l.push(yield);
+            l.push(yield);
+            l.push(yield);
+            return pyodide.toPy(l.map((x) => x.toString()));
```

Comment:
So, before this PR, proxies inside `l` is already destroyed so it will fail here I guess? Would you mind adding some comments about what these tests actually want to test? It was a bit hard for me to understand at first glance.

Reply:
Yeah. I'll add a comment.

Reply:
One funny thing here is that it creates a difference between generators that were created in JavaScript and then brought over to Python and generators that were accessed as an attribute from a `JsProxy`. There's a similar problem with promises. It creates an annoying inconsistency, not sure how best to deal with it.

I think also it would be good not to destroy proxies with `roundtrip=true` in a lot of cases...